Add virtual particle property destructor

gassmoeller opened the pull request:
Forgot to add these when making the pull request for particles.

bangerth commented on include/aspect/particle/property/interface.h:
```diff
@@ -70,6 +70,11 @@ namespace aspect
+           * Destructor. Made virtual so that derived classes can be created
+           * and destroyed through pointers to the base class.
+           */
+          virtual ~Interface () {}
```
We're generally not used to putting the implementation of functions into the class declaration, even if the function is empty. Please move it to the .cc file and then merge yourself.

gassmoeller commented:
Then I might as well clean up the other interfaces. Would you mind taking another quick look? I removed every not used constructor and moved the destructors to the cc files. It seems like I mixed this up at some point, and then copy-pasted it everywhere.
